R/knit_print.R

```diff
@@ -7,13 +7,37 @@
 #'
 #' @keywords internal
 #' @noRd
-knit_print.fontawesome <- function(x, ...) {
+knit_print.fontawesome <- function(x, ..., options, inline = FALSE) {
 
   # nocov start
 
   if (requireNamespace("knitr", quietly = TRUE)) {
-    knitr::knit_print(as.character(x), ...)
+    if (knitr::is_html_output()) {
+      NextMethod()
+    } else if (knitr::pandoc_to(c("latex", "beamer", "docx"))) {
+      if (!requireNamespace("rsvg", quietly = TRUE)) {
+        stop("Using fontawesome with non HTML output requires the rsvg package:\n",
+             " * It can be installed with `install.packages(\"rsvg\")`.",
+             call. = FALSE)
+      }
+      formats <- switch(knitr::pandoc_to(),
+                        beamer = ,
+                        latex = list(ext = ".pdf", renderer = rsvg::rsvg_pdf),
+                        docx = list(ext = ".png", renderer = rsvg::rsvg_png)
+      )
+      icon_file <- paste0(
+        options$fig.path,
+        basename(tempfile("fa-icon-", fileext = formats$ext))
+      )
+      if (!dir.exists(d <- dirname(icon_file))) dir.create(d, recursive = TRUE)
```

Comment:
Can you move the assignment out of the `if` statement?

Reply:
sorry Yihui's style starting to print on me. 😅  Sure I'll remove.